src/chainstate/burn/db/burndb.rs

```diff
@@ -22,132 +22,95 @@ use rusqlite::types::ToSql;
-pub struct HistoryRow {
-    pub txid: Txid,
-    pub vtxindex: u32,
-    pub block_height: u64,
-    pub op: u8,
-    pub burn_header_hash: BurnchainHeaderHash,
-    pub canonical: bool
-}
+use util::vrf::*;
+use util::secp256k1::MessageSignature;
+use util::hash::{to_hex, hex_bytes, Hash160, Sha512Trunc256Sum};
+use util::strings::StacksString;
 
-impl RowOrder for HistoryRow {
-    fn row_order() -> Vec<&'static str> {
-        vec!["txid","vtxindex","block_height","op","burn_header_hash","canonical"]
-    }
-}
+use net::neighbors::MAX_NEIGHBOR_BLOCK_DELAY;
 
-impl FromRow<HistoryRow> for HistoryRow {
-    fn from_row<'a>(row: &'a Row, index: usize) -> Result<HistoryRow, db_error> {
-        let txid = Txid::from_row(row, 0 + index)?;
-        let vtxindex : u32 = row.get(1 + index);
-        let block_height_i64 : i64 = row.get(2 + index);
-        let op : u8 = row.get(3 + index);
-        let burn_header_hash = BurnchainHeaderHash::from_row(row, 4 + index)?;
-        let canonical : bool = row.get(5 + index);
+use std::collections::HashMap;
 
-        if block_height_i64 < 0 {
-            return Err(db_error::ParseError);
-        }
+use vm::types::Value;
+use vm::representations::{ContractName, ClarityName};
 
-        let history_row = HistoryRow {
-            txid: txid,
-            vtxindex: vtxindex,
-            block_height: block_height_i64 as u64,
-            op: op,
-            burn_header_hash: burn_header_hash,
-            canonical: canonical
-        };
-        Ok(history_row)
-    }
-}
+const BLOCK_HEIGHT_MAX : u64 = ((1 as u64) << 63) - 1; 
+const SQLITE_ERROR_MSG : &'static str = "FATAL: failed to exeucte Sqlite database operation.  Aborting...";
 
-impl<A, K> From<&LeaderKeyRegisterOp<A, K>> for HistoryRow
-where
-    A: Address,
-    K: PublicKey
-{
-    fn from(leader_key: &LeaderKeyRegisterOp<A, K>) -> HistoryRow {
-        HistoryRow {
-            txid: leader_key.txid.clone(),
-            vtxindex: leader_key.vtxindex,
-            block_height: leader_key.block_number,
-            op: leader_key.op,
-            burn_header_hash: leader_key.burn_header_hash.clone(),
-            canonical: true
-        }
-    }
-}
+pub const REWARD_WINDOW_START : u64 = 144 * 15;
+pub const REWARD_WINDOW_END : u64 = 144 * 90 + REWARD_WINDOW_START;
 
-impl<A, K> From<&LeaderBlockCommitOp<A, K>> for HistoryRow 
-where
-    A: Address,
-    K: PublicKey
-{
-    fn from(block_commit: &LeaderBlockCommitOp<A, K>) -> HistoryRow {
-        HistoryRow {
-            txid: block_commit.txid.clone(),
-            vtxindex: block_commit.vtxindex,
-            block_height: block_commit.block_number,
-            op: block_commit.op,
-            burn_header_hash: block_commit.burn_header_hash.clone(),
-            canonical: true
-        }
+// for using BurnchainHeaderHash values as block hashes in a MARF
+impl From<BurnchainHeaderHash> for BlockHeaderHash {
+    fn from(bhh: BurnchainHeaderHash) -> BlockHeaderHash {
```

Comment:
This can just be `BlockHeaderHash(bhh.0)` and avoid an allocation and copy -- same goes for `From<BlockHeaderHash> for BurnchainHeaderHash`

Reply:
Addressed in a8f13b1b.